crates/phactory/api/src/crypto.rs

```diff
@@ -178,28 +215,39 @@ impl CertificateBody {
         msg_type: MessageType,
         sig_type: SignatureType,
         signature: &[u8],
-    ) -> Result<Vec<u8>, SignatureVerifyError> {
-        match sig_type {
+    ) -> Result<AccountId32, SignatureVerifyError> {
+        let signer = match sig_type {
             SignatureType::Ed25519 => {
-                recover::<sp_core::ed25519::Pair>(&self.pubkey, signature, msg)
+                recover::<sp_core::ed25519::Pair>(&self.pubkey, signature, msg)?.into()
             }
             SignatureType::Sr25519 => {
-                recover::<sp_core::sr25519::Pair>(&self.pubkey, signature, msg)
+                recover::<sp_core::sr25519::Pair>(&self.pubkey, signature, msg)?.into()
             }
-            SignatureType::Ecdsa => recover::<sp_core::ecdsa::Pair>(&self.pubkey, signature, msg),
+            SignatureType::Ecdsa => account_id_from_evm_pubkey(recover::<sp_core::ecdsa::Pair>(
+                &self.pubkey,
+                signature,
+                msg,
+            )?),
             SignatureType::Ed25519WrapBytes => {
                 let wrapped = wrap_bytes(msg);
-                recover::<sp_core::ed25519::Pair>(&self.pubkey, signature, &wrapped)
+                recover::<sp_core::ed25519::Pair>(&self.pubkey, signature, &wrapped)?.into()
             }
             SignatureType::Sr25519WrapBytes => {
                 let wrapped = wrap_bytes(msg);
-                recover::<sp_core::sr25519::Pair>(&self.pubkey, signature, &wrapped)
+                recover::<sp_core::sr25519::Pair>(&self.pubkey, signature, &wrapped)?.into()
             }
             SignatureType::EcdsaWrapBytes => {
                 let wrapped = wrap_bytes(msg);
-                recover::<sp_core::ecdsa::Pair>(&self.pubkey, signature, &wrapped)
+                account_id_from_evm_pubkey(recover::<sp_core::ecdsa::Pair>(
```

Comment:
Could you also sync the latest PR to this one. It looks like still the old impl.

Reply:
Yes, I am working on that.